server/server.js

```diff
@@ -1,13 +1,32 @@
 const path = require('path');
 const express = require('express');
+const tree = require('./tree');
 
 const app = express();
 
-app.use(express.static(path.resolve(__dirname, '../client')));
+app.use((req, res, next) => {
+  res.header('Access-Control-Allow-Origin', '*');
+  res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept');
+  next();
+});
+
+app.use(express.static(path.resolve(__dirname, '../client/public')));
+
 app.listen(process.env.npm_package_config_port);
 
-app.get('/', (req, res) => {
-  res.send('Connected');
+app.get('/api/tree', (req, res) => {
+  if (!Object.keys(req.query).length 
+    || !req.query.path.trim().length
+    || !req.query.path
+  ) {
+    res.status(204).send();
```

Comment:
I'm almost sure you can just `res.sendStatus(204)`
Should `return` here as well because otherwise you'll try to do multiple `send` into same response which would fail